src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public class RefExtensionMethodsTests : CSharpTestBase
+    {
+        [Fact]
+        public void ExtensionMethods_RValues_Ref()
+        {
+            var code = @"
+public static class Extensions
+{
+    public static void PrintValue(ref this int p)
```

Comment:
Also perhaps a test to make sure the `this` parameter is indeed `ref` or `ref readonly` as seen from inside of the method. - cannot pass is by ordinary `ref` when it is `ref readonly`, but can when it is `ref`. 
Also may be worth checking what semantic model says about the ref kind of the parameter. Just in case we have no assumptions that `this` is always byval inside the extensions' body.

---
In reply to: [120226187](https://github.com/dotnet/roslyn/pull/19703#discussion_r120226187) [](ancestors = 120226187)